### ISSUE TITLE
Improve widget borders

### ISSUE THREAD

**Issue by dalboris** [vgc/vgc#14]
I'm having trouble to style widget borders the way I'd like. Here is a bug entry to document these problems.

Initially, I noticed that I had 2px borders of the Python console instead of the desired 1px:

![01-qss-border-1](https://user-images.githubusercontent.com/4809739/41151876-5578cbae-6b12-11e8-83fa-682259521e17.png)

Since these come from borders of internal widgets I don't have full control over, I thought it could be a good idea in this case to use a ["CSS reset"](https://www.webpagefx.com/blog/web-design/should-you-reset-your-css/) approach, implemented as `QWidget { border = 0px }`, to set all borders to zero by default. This partially worked for the Console, but has the undesired side effect to remove borders on other internal widgets where they are desired (e.g., QColorDialog), and also causes a surprising shrinking of the QFileDialog button icons, for which I have found no workaround:

![01-qss-border-2](https://user-images.githubusercontent.com/4809739/41152337-9f54bb6a-6b13-11e8-92b8-73ccd10d5155.png)

The latter issues are worse than the former, thus at the moment I decided to leave `/* border: 0px */` commented. But ideally we should find a fix to get all borders the way we'd like! This may involve implementing my own color dialog and file dialog, or potentially ugly hacks to apply style to internal widgets (which could break when upgrading Qt).

**Comment by dalboris** [comment]
Related: #13

**Comment by Robot-Fromage** [comment]
If i may, this is just an idea but since you're using QSS stylesheet you can set the desired style for your specific QWidget and then prevent inheriting these styles with 'revert' or 'default' property, without extra work.
For example if your widget is named with setObjectName:

```
/* MyWidget has a custom style */
#MyWidget
{
    border: 1px solid black;
}

/* Direct or Indirect childs of MyWidget shouldn't inherit the parent custom style */
#MyWidget QWidget
{
    all: revert;
    /* You could also try: */
    /* all: default; */
    /* border: revert; */
    /* border: default; */
}
```

**Comment by dalboris** [comment]
Thanks, I didn't know about that, this seems quite useful!

**Comment by dalboris** [comment]
Now, it looks like this:

![image](https://user-images.githubusercontent.com/4809739/63227080-0bba3380-c1e3-11e9-9ba1-23c2bcb3e832.png)

The main idea was that instead of using borders, we can use different background colors to differentiate between the different areas of the application. This fixes the problem of duplicated borders, and it gives the application a slightly more modern look in my opinion.

One of the easy fix was to add a `border: 0px` attribute to the toolbar, I'm not sure why it wasn't the case yet, this was the cause of the 2px instead of 1px border between the toolbar and the console.

I kept non-0px separators between docks since those are currently necessary for resizing the docks. I actually increased the size of the separators to 5px to make the docks easier to resize, especially when using a pen tablet. But I changed their color to be the same as the menubar and console background, so that they look less like borders between the docks, and importantly, it doesn't add a border between the console and the drawing area. Ideally, I'd still prefer to have 0px-separator, but unfortunately, at least on Windows, this makes the docks impossible to resize.

I changed the "console left margin" background to be the same as the actual console background, as it makes the app structure easier to understand at a glance. Now it's way clearer that the console margin is part of the console, rather than the toolbar.

Finally, I fine-tuned the font size, the colors and spacing of the menus, and vertical centering of text in buttons, line edits, and combo boxes.
